### PR TITLE
Mocks for sleep in tests

### PR DESCRIPTION
**Pull request by Obs01ete**
## Description

`time.sleep`s are mocked during testing.

## Motivation and Context

Tests were very slow (6 min) due to sleeps.

## Types of changes

What types of changes does your code introduce? Put an `x` in all the boxes that apply:

- [x] Tests improvement

### CODE REVIEW

File: .github/workflows/pytest_apps.yml [hunk 1]
```diff
@@ -33,7 +33,7 @@ jobs:
     - name: Install dependencies
       run: |
         python -m pip install --upgrade pip
-        pip install flake8 pytest
+        pip install flake8 pytest mock
```
Comment (lightaime) [inline .github/workflows/pytest_apps.yml:36]: Should we add `mock` to `pyproject.toml` as well? Maybe in the future we want to structure `pyproject.toml` a bit for dev mode.

**Review by lightaime — APPROVED** [review]
LGTM!

File: examples/test/test_single_agent.py [hunk 2]
```diff
@@ -36,9 +37,12 @@ def test_misalignment_single_agent(model):
     examples.misalignment.single_agent.main(model=model)
 
 
-@parametrize
+@pytest.mark.parametrize('model', [
+    ModelType.STUB,
+])
 def test_evaluation_single_agent(model):
-    examples.evaluation.single_agent.main(model=model)
+    with patch('time.sleep', return_value=None):
```
Comment (lightaime) [inline examples/test/test_single_agent.py:44]: Where is the `time.sleep` in `examples.evaluation.single_agent`?

**Review by lightaime — CHANGES_REQUESTED** [review]

File: examples/test/test_single_agent.py [hunk 3]
```diff
@@ -55,4 +59,5 @@ def test_code_generate_metadata(model):
         #              marks=[pytest.mark.model_backend, pytest.mark.slow]),
     ])
 def test_code_task_generation(model):
-    examples.code.task_generation.main(model=model)
+    with patch('time.sleep', return_value=None):
+        examples.code.task_generation.main(model=model)
```
Comment (lightaime) [inline examples/test/test_single_agent.py:63]: Where is the `time.sleep` in `examples.code.task_generation`?

File: .github/workflows/pytest_apps.yml (hunk 1 above)
Comment (Obs01ete) [reply .github/workflows/pytest_apps.yml:36]: Nope. This change is for tests only. `mock` is of no use for the package.

File: examples/test/test_single_agent.py (hunk 2 above)
Comment (Obs01ete) [reply examples/test/test_single_agent.py:44]: Thanks for spotting. I did not find it, but assumed that it is there somewhere :) Since this example is very slow.

Comment (Obs01ete) [reply examples/test/test_single_agent.py]: Reverted

File: examples/test/test_single_agent.py (hunk 3 above)
Comment (Obs01ete) [reply examples/test/test_single_agent.py]: Reverted

File: .github/workflows/pytest_apps.yml (hunk 1 above)
Comment (lightaime) [reply .github/workflows/pytest_apps.yml:36]: I mean we should create a group for the test dependencies in `pyproject.toml` like `[tool.poetry.group.test.dependencies]`.

File: examples/test/test_single_agent.py [hunk 4]
```diff
@@ -36,7 +36,9 @@ def test_misalignment_single_agent(model):
     examples.misalignment.single_agent.main(model=model)
 
 
-@parametrize
+@pytest.mark.parametrize('model', [
+    ModelType.STUB,
+])
```
Comment (lightaime) [inline examples/test/test_single_agent.py:41]: Why this change is necessary?

Comment (Obs01ete) [reply examples/test/test_single_agent.py:41]: It was too slow with the real model. We do not check the semantics anyway. Stub will do the trick for the testing purpose.

Comment (lightaime) [reply examples/test/test_single_agent.py:41]: Sure

**Review by lightaime — APPROVED** [review]
